src/main/java/org/broadinstitute/dsm/route/PatchRoute.java

```diff
@@ -263,6 +272,27 @@ else if (participantDataId != null) {
                         }
                         return new Result(200, new GsonBuilder().serializeNulls().create().toJson(map));
                     }
+                    else if (Patch.DDP_PARTICIPANT_ID.equals(patch.getParent())) {
+                        //new additional value for pt which is not in ddp_participant and ddp_participant_record table yet
+                        DDPInstance ddpInstance = DDPInstance.getDDPInstance(patch.getRealm());
+                        String participantId = inTransaction(conn -> {
+                            String newParticipantId = MedicalRecordUtil.writeParticipantIntoDB(conn, patch.getParentId(), ddpInstance.getDdpInstanceId(), 0, "", patch.getUser());
+                            MedicalRecordUtil.writeNewRecordIntoDb(conn, DDPMedicalRecordDataRequest.SQL_INSERT_PARTICIPANT_RECORD, patch.getParentId(), ddpInstance.getDdpInstanceId());
+                            return newParticipantId;
+                        });
+                        if (StringUtils.isNotBlank(participantId)) {
```

Comment:
I always prefer to use `Optional`(Optional.ofNullable) as a return type from method which returns single result or empty `List` if it returns collection of records when interaction happens with DB, with `Optional` and empty `List` we can provide more cleaner code and better way to avoid NPEs.

Reply:
feel free to change that